### PR TITLE
[bitnami/postgresql] Make existing secret key names choosable

### PR DESCRIPTION
**Pull request by TobiMichael96**
Signed-off-by: Tobias Effner <tobias.effner@rackspace.com>

### Make existing secret key names choosable

This PR will enable developers to choose their own secret key name in an existing secret.

### Benefits

It adds freedom in the existing secret to match existing patterns in the developers code.

### Possible drawbacks

None.

### Applicable issues

None.

### Additional information

I kept the logic the chart uses to prioritize the different passwords.

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/postgresql/Chart.yaml [hunk 1]
```diff
@@ -26,4 +26,4 @@ name: postgresql
 sources:
   - https://github.com/bitnami/bitnami-docker-postgresql
   - https://www.postgresql.org/
-version: 11.2.6
+version: 12.2.6
```
Comment (fmulero) [inline bitnami/postgresql/Chart.yaml:29]: Why this version? At the moment it has a breaking change so it should be 12.0.0 not 12.2.6. Please have a look to [semver](https://semver.org/)

File: bitnami/postgresql/README.md [hunk 2]
```diff
@@ -2,12 +2,14 @@
 
 # PostgreSQL packaged by Bitnami
 
-PostgreSQL (Postgres) is an open source object-relational database known for reliability and data integrity. ACID-compliant, it supports foreign keys, joins, views, triggers and stored procedures.
+PostgreSQL (Postgres) is an open source object-relational database known for reliability and data integrity.
```
Comment (fmulero) [inline bitnami/postgresql/README.md:5]: Is there any reason to break the lines? Please follow the same conventions we already have. Could you also let bitnami-bot run the readme generator in your repository? We are not able to merge any external PR with [failures in the pipeline](https://github.com/bitnami/charts/runs/6550236832?check_suite_focus=true)

File: bitnami/postgresql/values.yaml [hunk 3]
```diff
@@ -19,14 +19,23 @@ global:
...
       username: ""
       password: ""
       database: ""
-      existingSecret: ""
+      existingSecret:
+        enabled: false
+        secretName: name-of-secret
+        adminPasswordKey: admin-password
+        userPasswordKey: user-password
+        replicationPasswordKey: replication-password
```
Comment (fmulero) [inline bitnami/postgresql/values.yaml:38]: In this case we can add this new feature without breaking previous versions. For example:

```suggestion
      existingSecret: ""
      secretKeys:
        adminPasswordKey: admin-password
        userPasswordKey: user-password
        replicationPasswordKey: replication-password
```

Keep using `existingSecret` as name for the secret and adding a new parameter for the different keys. With this approach user don't need to change their values files to keep working and we don't need to upgrade all charts depending on postgresql

**Review by fmulero — CHANGES_REQUESTED** [review]
Hi @TobiMichael96 Thanks a lot for your contribution.

I left some comments. Could you take  them a look?

File: bitnami/postgresql/values.yaml (hunk 3 above)
Comment (TobiMichael96) [reply bitnami/postgresql/values.yaml:38]: That's a good idea, thanks for that! I'll try to implement that.

File: bitnami/postgresql/README.md (hunk 2 above)
Comment (TobiMichael96) [reply bitnami/postgresql/README.md:5]: My IDE did that, I will revert that change.

File: bitnami/postgresql/Chart.yaml (hunk 1 above)
Comment (TobiMichael96) [reply bitnami/postgresql/Chart.yaml:29]: You are totally right.

**Comment by TobiMichael96** [comment]
> Hi @TobiMichael96 Thanks a lot for your contribution.
> 
> I left some comments. Could you take them a look?

I checked all your comments. Please check if it's fine now. :)

File: bitnami/postgresql/values.yaml [hunk 4]
```diff
@@ -131,10 +138,17 @@ auth:
...
-  ## @param auth.existingSecret Name of existing secret to use for PostgreSQL credentials. The secret must contain the keys `postgres-password` (which is the password for "postgres" admin user), `password` (which is the password for the custom user to create when `auth.username` is set) and `replication-password` (which is the password for replication user). `auth.postgresPassword`, `auth.password`, and `auth.replicationPassword` will be ignored and picked up from this secret. The secret might also contains the key `ldap-password` if LDAP is enabled. `ldap.bind_password` will be ignored and picked from this secret in this case.
-  ## The value is evaluated as a template.
+  ## @param auth.existingSecret Name of existing secret to use for PostgreSQL credentials. `auth.postgresPassword`, `auth.password`, and `auth.replicationPassword` will be ignored and picked up from this secret. The secret might also contains the key `ldap-password` if LDAP is enabled. `ldap.bind_password` will be ignored and picked from this secret in this case.
   ##
   existingSecret: ""
+  ## @param auth.secretKeys.adminPasswordKey Name of key in existing secret to use for PostgreSQL credentials.
```
Comment (fmulero) [inline bitnami/postgresql/values.yaml:144]: `auth.secretKeys` is used only when `auth.existingSecret` is set. Could you add something in the description to ensure that's clear for other users?

**Review by fmulero — CHANGES_REQUESTED** [review]
Hi @TobiMichael96 thanks for sorting out so fast. I was doing some tests and I added some additional comments.

File: bitnami/postgresql/values.yaml [hunk 5]
```diff
@@ -19,14 +19,21 @@ global:
...
     auth:
       postgresPassword: ""
       username: ""
       password: ""
       database: ""
       existingSecret: ""
+      secretKeys:
+        adminPasswordKey: admin-password
+        userPasswordKey: user-password
+        replicationPasswordKey: replication-password
```
[the remark this answers is not in the document]
Comment (TobiMichael96) [reply bitnami/postgresql/values.yaml]: See latest commit: https://github.com/bitnami/charts/pull/10347/commits/f6384b7857f15e3169e0bc4587c5d39012fab612

File: bitnami/postgresql/values.yaml [hunk 6]
```diff
@@ -19,14 +19,21 @@ global:
     ## @param global.postgresql.auth.username Name for a custom user to create (overrides `auth.username`)
     ## @param global.postgresql.auth.password Password for the custom user to create (overrides `auth.password`)
     ## @param global.postgresql.auth.database Name for a custom database to create (overrides `auth.database`)
-    ## @param global.postgresql.auth.existingSecret Name of existing secret to use for PostgreSQL credentials (overrides `auth.existingSecret`)
+    ## @param global.postgresql.auth.existingSecret Name of existing secret to use for PostgreSQL credentials (overrides `auth.existingSecret`).
+    ## @param global.postgresql.auth.secretKeys.adminPasswordKey Name of key in existing secret to use for PostgreSQL credentials.
```
[the remark this answers is not in the document]
Comment (TobiMichael96) [reply bitnami/postgresql/values.yaml]: See latest commit: https://github.com/bitnami/charts/pull/10347/commits/f6384b7857f15e3169e0bc4587c5d39012fab612

File: bitnami/postgresql/values.yaml (hunk 4 above)
Comment (TobiMichael96) [reply bitnami/postgresql/values.yaml]: See latest commit: https://github.com/bitnami/charts/pull/10347/commits/f6384b7857f15e3169e0bc4587c5d39012fab612

File: bitnami/postgresql/values.yaml [hunk 7]
```diff
@@ -131,10 +138,17 @@ auth:
...
   ##
   existingSecret: ""
+  ## @param auth.secretKeys.adminPasswordKey Name of key in existing secret to use for PostgreSQL credentials.
+  ## @param auth.secretKeys.userPasswordKey Name of key in existing secret to use for PostgreSQL credentials.
+  ## @param auth.secretKeys.replicationPasswordKey Name of key in existing secret to use for PostgreSQL credentials.
+  ##
+  secretKeys:
+    adminPasswordKey: admin-password
+    userPasswordKey: user-password
+    replicationPasswordKey: replication-password
```
[the remark this answers is not in the document]
Comment (TobiMichael96) [reply bitnami/postgresql/values.yaml]: See latest commit: https://github.com/bitnami/charts/pull/10347/commits/f6384b7857f15e3169e0bc4587c5d39012fab612

**Comment by TobiMichael96** [comment]
> Hi @TobiMichael96 thanks for sorting out so fast. I was doing some tests and I added some additional comments.

Thanks for your fast review and your suggestions! :) I think it looks better now. Please have a look at my update.

**Review by fmulero — DISMISSED** [review]
LGTM, great work @TobiMichael96. I'll run some extra tests , if everything works fine, your changes will be automatically merged to master

**Comment by TobiMichael96** [comment]
> LGTM, great work @TobiMichael96. I'll run some extra tests , if everything works fine, your changes will be automatically merged to master

Thanks for the review and your input, that helped me a lot. :)

**Pull request merged**